dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer/ApplyCouponToCartTest.php

```diff
@@ -0,0 +1,419 @@
+        $this->quoteResource->save($this->quote);
+        $queryHeaders = $this->prepareAuthorizationHeaders('customer@example.com', 'password');
+        $query = $this->prepareAddCouponRequestQuery($maskedQuoteId, $couponCode);
+
+        self::expectExceptionMessage('The current user cannot perform operations on cart "' . $maskedQuoteId . '"');
+        $this->graphQlQuery($query, [], '', $queryHeaders);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Customer/_files/customer.php
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_simple_product_saved.php
+     * @expectedException \Exception
+     * @expectedExceptionMessage The coupon code isn't valid. Verify the code and try again.
+     */
+    public function testApplyNonExistentCouponToCart()
+    {
+        $couponCode = '1%q#f5';
+        $this->quoteResource->load(
+            $this->quote,
+            'test_order_with_simple_product_without_address',
+            'reserved_order_id'
+        );
+        $maskedQuoteId = $this->quoteIdToMaskedId->execute((int)$this->quote->getId());
+        $this->quote->setCustomerId(1);
+        $this->quoteResource->save($this->quote);
+        $queryHeaders = $this->prepareAuthorizationHeaders('customer@example.com', 'password');
+        $query = $this->prepareAddCouponRequestQuery($maskedQuoteId, $couponCode);
+
+        $this->graphQlQuery($query, [], '', $queryHeaders);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Customer/_files/customer.php
+     * @magentoApiDataFixture Magento/SalesRule/_files/coupon_code_with_wildcard.php
+     * @expectedException \Exception
+     */
+    public function testApplyCouponToNonExistentCart()
+    {
+        $couponCode = '2?ds5!2d';
+        $maskedQuoteId = '1hk3y1842h1n';
+        $this->quote->setCustomerId(1);
+        $this->quoteResource->save($this->quote);
+        $queryHeaders = $this->prepareAuthorizationHeaders('customer@example.com', 'password');
+        $query = $this->prepareAddCouponRequestQuery($maskedQuoteId, $couponCode);
+
+        self::expectExceptionMessage('Could not find a cart with ID "' . $maskedQuoteId . '"');
+        $this->graphQlQuery($query, [], '', $queryHeaders);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Customer/_files/customer.php
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_simple_product_saved.php
+     * @magentoApiDataFixture Magento/SalesRule/_files/coupon_code_with_wildcard.php
+     * @expectedException \Exception
+     * @expectedExceptionMessage The coupon code isn't valid. Verify the code and try again.
+     */
+    public function testApplyExpiredCoupon()
+    {
+        $couponCode = '2?ds5!2d';
+        $this->quoteResource->load(
+            $this->quote,
+            'test_order_with_simple_product_without_address',
+            'reserved_order_id'
+        );
+        $maskedQuoteId = $this->quoteIdToMaskedId->execute((int)$this->quote->getId());
```

Comment:
Looks like we can avoid manipulating with QuoteResource (etc..) if we will reuse current fixtures
```
/**
     * @magentoApiDataFixture Magento/Customer/_files/customer.php
     * @magentoApiDataFixture Magento/Catalog/_files/product_simple.php
     * @magentoApiDataFixture Magento/GraphQl/Quote/_files/customer/create_empty_cart.php
     * @magentoApiDataFixture Magento/GraphQl/Quote/_files/add_simple_product.php
     * @magentoApiDataFixture Magento/GraphQl/Quote/_files/set_new_shipping_address.php
     */
```